channeldb/witness_cache.go

```diff
@@ -77,6 +78,24 @@ type witnessEntry struct {
 	witness []byte
 }
 
+// AddSha256Witnesses adds a batch of new sha256 preimages into the witness
+// cache. This is an alias for AddWitnesses that uses Sha256HashWitness as the
+// preimages' witness type.
+func (w *WitnessCache) AddSha256Witnesses(preimages ...lntypes.Preimage) error {
+	// Optimistically compute the preimages' hashes before attempting to
+	// start the db transaction.
+	entries := make([]witnessEntry, 0, len(preimages))
+	for i := range preimages {
+		hash := preimages[i].Hash()
```

Comment:
any reason not to use `for _, p := range preimages {`?

Reply:
Indeed because each `p[:]` will point to the same preimage due to range var binding. This was the bug I had initially ;)